src/wallet/rpcdump.cpp

```diff
@@ -302,8 +302,7 @@ UniValue importprunedfunds(const UniValue& params, bool fHelp)
     LOCK2(cs_main, pwalletMain->cs_wallet);
 
     if (pwalletMain->IsMine(tx)) {
-        CWalletDB walletdb(pwalletMain->strWalletFile, "r+", false);
```

Comment:
Indeed this does change the default. There doesn't seem to be a good reason not to flush here.